modules/dists/SparseBlockDist.chpl

```diff
@@ -485,7 +485,7 @@ class SparseBlockArr: BaseSparseArr {
     // Ensure it is legal to invoke the standalone iterator
     // on locA.myElems below.
     __primitive("resolves",
-                locArr[locArrDom.low]!.myElems._value.these(tag)) {
+                locArr[locArrDom.lowBound]!.myElems._value.these(tag)) {
```

Comment:
@e-kayrakli : In updating this, I tried to write a simple test that demonstrated the problem which existed before and got a "can't resolve type due to recursive call" error for the following simple program (which also happens without the change):

```chapel
use BlockDist;

var A = newBlockArr({1..10 by 2, 1..10 by 2}, real);
var DS: sparse subdomain(A.domain);
var AS: [DS] real;

forall a in AS do
  writeln(a);
```

Do you see the same?  Could it be that we have no tests of standalone parallel iteration over sparse block arrays?

If I comment out the primitive, it seems to work as expected.

Reply:
(though I'm now noticing that @dlongnecke-cray and @vasslitvinov have added this since your work, so probably I should be asking them).

Reply:
Hmm that snippet works for me on `main`?

Reply:
The resolves primitive has some bugs that I haven't gotten around to fixing yet, so definitely feel free to file an issue or something if that's the problem.

Reply:
> Hmm that snippet works for me on main?

Huh... switching to a new workspace, it's now working for me as well.  I'm not sure what was up (I long since changed branches on my original workspace).  Thanks for double-checking me!

Reply:
Ah, I got it:  I get the infinite recursion for CHPL_COMM=none, but not for CHPL_COMM!=none.  By any chance were you testing with `CHPL_COMM!=none`?

Reply:
Ah, you're right. I got `$CHPL_HOME/modules/dists/SparseBlockDist.chpl:488: error: called recursively at this point` with COMM=none.

Reply:
Thanks for confirming!  I'm not entirely clear what the point of that where clause is, and no tests seem to rely on it, so I'm inclined to comment it out for this PR and leave it as "future work" for someone to enable it in a way that works with the new test added to this PR.  Let me know if you disagree with that approach.

Reply:
I kind of blew off @dlongnecke-cray 's comment above, figuring "We didn't have any tests that exercised this, so it's probably a problem that always existed."  But... it seems that he's right.  Specifically, when I go in and annotate the call with some printing out of types being invoked on, there is no recursion as is being complained about (where I thought it seemed conceivable that we had a bug in which the layout we used for CHPL_COMM=none sparse block distributions might change).  And if I change it back to an old-style "method resolves" call, it works as expected.  So... that's disconcerting.

That gives me a path forward for this PR (switch back to "method resolves", but leaves me unnerved about the stability of "resolves" and what other errors could be lurking in our code once the class hierarchy gets complex enough.  I'm also not sure how to capture this as an issue b/c I think reproducing the complexity of the class hierarchies that implement arrays is pretty significant task.  David, any thoughts there?

Reply:
Oh wait wait wait... I had a realization over dinner.  I think it's not that the new primitive is giving incorrect results, it's that it's not pruning aggressively enough.  Specifically, I'm guessing that to answer the question "Can I resolve this?" it's looking for some/all methods with the given name and trying to resolve them, one of which is the call itself.  But, presumably, it shouldn't get that far because the dispatching types aren't correct.  So maybe I can create a simple reproducer for this after all.  I'll give it a shot.

Reply:
Huh... Nope, not able to put together a reproducer easily...

Reply:
A reproducer would be helpful, but I can also make a note privately that points to this PR.

In the meantime, could reverting the primitive call to `__primitive("method call resolves")` or `__primitive("method call and fn resolves")` help? The diffs in #19029 can help there.

Reply:
Doh, I didn't read the diffs, seems like you already reverted.

To comment further: I think whether or not the PRIM_RESOLVES resolves some or all of the candidate is still up for discussion, and this seems to be one of those cases where we rely on not resolving the candidate body for the correct behavior.

> Specifically, I'm guessing that to answer the question "Can I resolve this?" it's looking for some/all methods with the given name and trying to resolve them, one of which is the call itself. But, presumably, it shouldn't get that far because the dispatching types aren't correct.

So something along these lines is still correct.

This seems like it could inform the design of the user-facing "resolves" in that if something doesn't resolve, it would be nice to know why (instead of just silently folding away due to a param `false`).

Reply:
I forked this off to https://github.com/chapel-lang/chapel/pull/19820 with instructions for how to reproduce on `main` since I wasn't successful in getting it into a standalone reproducer.